### ISSUE TITLE
Make "block and report" a feature of the report option, rather than the block option

### ISSUE THREAD

**Issue by Lunaphied** [feditext/feditext#434]
**Is your feature request related to a problem? Please describe.**
When I find myself reporting I usually also wish to block the user, so I click report, expecting the option to block along with reporting. Confusing to me, this option is actually part of the *blocking* flow rather than the reporting flow, and I feel like it's more common to handle this the opposite way. 

**Describe the solution you'd like**
Move the "block and report" option pop-up to the report functionality rather than the block functionality.

**Describe alternatives you've considered**
Keeping the existing behavior and simply adding another pop-up for the report flow with the option. - This would be fine

**Comment by VyrCossont** [comment]
Yeah, this sounds useful. I've also wanted to include adding notes to the block/mute flow on platforms that support it.
